libvast/vast/concept/parseable/vast/schema.hpp

```diff
@@ -122,19 +123,26 @@ struct symbol_resolver {
         } else if (it->name == "+>") {
           acc = priority_merge(acc, *rhs, merge_policy::prefer_right);
         } else if (it->name == "-") {
-          auto name = rhs->fields[0].name;
-          auto del = std::find_if(acc.fields.begin(), acc.fields.end(),
-                                  [&](auto& f) { return f.name == name; });
-          if (del == acc.fields.end())
+          std::vector<std::string_view> path;
+          for (auto& f : rhs->fields)
+            path.emplace_back(f.name);
+          if (!remove_field(acc, path)) {
+            auto fname = detail::join(path, ".");
             return caf::make_error(ec::parse_error,
                                    fmt::format("trying to delete non-existing "
                                                "field {} from type {}",
-                                               rhs->name(), to_string(acc)));
-          acc.fields.erase(del);
+                                               fname, to_string(acc)));
```

Comment:
Instead of introducing the `fname` variable, consider using `fmt::join(path, ".")` as an argument to the `fmt::format` call over `fname`. This avoids an unnecessary copy and some allocations, and removes the need to include `vast/detail/string.hpp`.